src/webnn/native/dml/UtilsDML.cpp

```diff
@@ -0,0 +1,118 @@
+// Copyright 2021 The WebNN-native Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "UtilsDML.h"
+
+namespace webnn::native::dml {
+
+    bool IsWarpAdapter(IDXGIAdapter1* pAdapter) {
+        DXGI_ADAPTER_DESC1 pDesc;
+        ASSERT_IF_FAILED(pAdapter->GetDesc1(&pDesc));
+        // See here for documentation on filtering WARP adapter:
+        // https://docs.microsoft.com/en-us/windows/desktop/direct3ddxgi/d3d10-graphics-programming-guide-dxgi#new-info-about-enumerating-adapters-for-windows-8
+        auto isBasicRenderDriverVendorId = pDesc.VendorId == 0x1414;
```

Comment:
unit: do not use `auto` for integral types, it just obfuscates the code and more so, defeats the point of using a statically typed language.